torch/_higher_order_ops/invoke.py

```diff
@@ -0,0 +1,88 @@
+import functools
```

Comment:
We can land this in its own PR, with its own tests.

Reply:
SGTM. Bikeshedding, I hate this name...